smoke_test/tests/authenticated/orchestration_tests.py

```diff
@@ -0,0 +1,67 @@
+import re
+import requests
+import json
+
+from uuid import uuid4
+from functools import cache
+from requests import Response
+from unittest import TestCase
+from urllib.parse import urljoin
+
+
+class OrchestrationTests(TestCase):
+  ORCHESTRATION_HOST = None
+  WORKSPACE_NAMESPACE = None
+  WORKSPACE_NAME = None
+  ACCESS_TOKEN = None
+
+  @staticmethod
+  def build_orchestration_url(path: str) -> str:
+    assert OrchestrationTests.ORCHESTRATION_HOST, "ERROR - OrchestrationSmokeTestCase.ORCHESTRATION_HOST not properly set"
+    if re.match(r"^\s*https?://", OrchestrationTests.ORCHESTRATION_HOST):
+      return urljoin(OrchestrationTests.ORCHESTRATION_HOST, path)
+    else:
+      return urljoin(f"https://{OrchestrationTests.ORCHESTRATION_HOST}", path)
+
+  @staticmethod
+  @cache
+  def call_orchestration(path: str) -> Response:
+    """Function is memoized so that we only make the call once"""
+    access_token = OrchestrationTests.ACCESS_TOKEN
+    oidc_headers = {
+      "Authorization": f"Bearer {access_token}",
+      "OIDC_access_token": access_token,
+      "OIDC_CLAIM_user_id": "cwds_orchestration_test_id",
+      "OIDC_CLAIM_expires_in": "10000",
+      "OIDC_CLAIM_email": "cwds.orchestration.test@example.com"
+    }
+    return requests.get(OrchestrationTests.build_orchestration_url(path), headers=oidc_headers)
+
+  @staticmethod
+  def workspace_path() -> str:
+    """Generate the path for the configured workspace"""
+    return f"/api/workspaces/{OrchestrationTests.WORKSPACE_NAMESPACE}/{OrchestrationTests.WORKSPACE_NAME}"
+
+  @staticmethod
+  def job_path(job_id) -> str:
+    """Generate the url to get status of a given job id"""
+    return f"{OrchestrationTests.workspace_path()}/importJob/{job_id}"
+
+  @staticmethod
+  def job_listing_path() -> str:
+    """Generate the path to list jobs in a given workspace"""
+    return f"{OrchestrationTests.workspace_path()}/importJob?runningOnly=true"
+
+  def test_random_job_status_is_404(self):
+    """Call the job-status url; it should return 404 since we are specifying a random job id"""
+    response = OrchestrationTests.call_orchestration(self.job_path(uuid4()))
+    self.assertEqual(response.status_code, 404,
+                     f"Job Status HTTP Status is not 404: {response.text}")
+
+  def test_job_listing_status_is_200_with_array(self):
+    """Call the job-listing url; it should return 200 and the response should be an array"""
+    response = OrchestrationTests.call_orchestration(self.job_listing_path())
+    self.assertEqual(response.status_code, 200,
+                     f"Job Listing HTTP Status is not 200: {response.text}")
+    job_list = json.loads(response.text)
```

Comment:
Nit: Requests has a convenience method for this

```suggestion
    job_list = response.json()
```

https://requests.readthedocs.io/en/latest/api/#requests.Response.json

Reply:
Removed.